source/extensions/dynamic_modules/sdk/rust/src/lib.rs

```diff
@@ -1661,13 +1661,13 @@ impl EnvoyHttpFilter for EnvoyHttpFilterImpl {
     }
   }
 
-  fn new_scheduler(&self) -> Box<dyn EnvoyHttpFilterScheduler> {
+  fn new_scheduler(&self) -> impl EnvoyHttpFilterScheduler + 'static {
```

Comment:
Hmm static lifetime ...

Reply:
Yeah - the default is `'_`, meaning borrowing lifetime of `EnvoyHttpFilter`. Since we're returning an owned object, `'static` is appropriate - I think...

Reply:
BTW while I don't know rust that deeply yet, I found this thread on `Box` having static lifetime by default. So in other words, this PR shouldn't be changing the lifetime of `new_scheduler`.

https://users.rust-lang.org/t/why-does-box-dyn-trait-introduce-a-static-lifetime/114608